### PR TITLE
[stl_killer_1]:stl list learning

### PR DESCRIPTION
**Pull request by Vancool**
### What

- brief: stl list learn
- author: vancool, goodfanqie, yeshihai

### Directory Specification

```
.
├── README.md
├── goodfanqie
│   ├── README.md
│   ├── doc
│   │   ├── README.md
│   │   └── task_1
│   │       └── list_learn.md
│   └── src
│       ├── task_0
│       │   └── main.cpp
│       └── task_1
│           └── stl_list.h
├── vancool
│   ├── README.md
│   └── doc
│       └── task_1
│           ├── list.md
│           └── list_pics
│               ├── class_rels.png
│               └── struct.png
└── yeshihai
    ├── doc
    │   └── task_1
    │       └── README.md
    └── src
        └── task_1
            └── main.cpp
```

### What changes are included in this PR?

learning gcc stl list & add note details

### Are these changes tested?

Manual test pass

### CODE REVIEW

File: stl/stl_killer/goodfanqie/src/task_0/main.cpp
Comment (Light-City) [inline stl/stl_killer/goodfanqie/src/task_0/main.cpp:1]: this is you test ?

File: stl/stl_killer/goodfanqie/src/task_1/stl_list.h
Comment (Light-City) [inline stl/stl_killer/goodfanqie/src/task_1/stl_list.h:1]: why move gcc source to here?

File: stl/stl_killer/vancool/src/main.cpp
Comment (Light-City) [inline stl/stl_killer/vancool/src/main.cpp:1]: no test.

File: stl/stl_killer/vancool/doc/task_1/List d5eb667438694baa958d19ddf522c9d8/Untitled 1.png
Comment (Light-City) [inline stl/stl_killer/vancool/doc/task_1/List d5eb667438694baa958d19ddf522c9d8/Untitled 1.png:1]: Directory does not conform to specification
List d5eb667438694baa958d19ddf522c9d8/Untitled 1.png

1. List d5eb667438694baa958d19ddf522c9d8 ？？？
2. Untitled 1.png？？

The directory name should make sense and follow lowercase underscore

File: stl/stl_killer/vancool/src/main.cpp
Comment (Vancool) [reply stl/stl_killer/vancool/src/main.cpp:1]: The empty file is removed.

Comment (Vancool) [reply stl/stl_killer/vancool/src/main.cpp:1]: > no test.

The empty file is removed.

Comment (Vancool) [reply stl/stl_killer/vancool/src/main.cpp:1]: The empty file is removed.

File: stl/stl_killer/vancool/doc/task_1/List d5eb667438694baa958d19ddf522c9d8/Untitled 1.png
Comment (Vancool) [reply stl/stl_killer/vancool/doc/task_1/List d5eb667438694baa958d19ddf522c9d8/Untitled 1.png:1]: Fix with renaming the pics.

File: stl/stl_killer/goodfanqie/src/task_1/stl_list.h
Comment (Vancool) [reply stl/stl_killer/goodfanqie/src/task_1/stl_list.h:1]: > why move gcc source to here?

update by removing member.

File: stl/stl_killer/goodfanqie/src/task_0/main.cpp
Comment (Vancool) [reply stl/stl_killer/goodfanqie/src/task_0/main.cpp:1]: update by removing member.

File: stl/stl_killer/yeshihai/src/task_1/main.cpp
Comment (Light-City) [inline stl/stl_killer/yeshihai/src/task_1/main.cpp:1]: this is empty test?

**Comment by Light-City** [comment]
reject no changes.

**Pull request closed without merging**